### PR TITLE
sssd: 1.16.5 -> 2.6.0, fix broken build

### PR DESCRIPTION
**Pull request by illustris**
###### Motivation for this change
SSSD has been broken on unstable for a while:
Build Status for sssd.x86_64-linux on master
✖ (Failed) sssd-1.16.5 from 2021-10-21 - https://hydra.nixos.org/build/155825548

It has been broken since https://github.com/NixOS/nixpkgs/commit/616690dc6439e315da12523f3cc1602939f6364e

###### Things done

- build dnsutils with GSSAPI by default (needed by SSSD)
- bump SSSD to 2.6.0
- clean up sssd-ldap test to make building easier

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/servers/dns/bind/default.nix [hunk 1]
```diff
@@ -1,6 +1,7 @@
 { config, stdenv, lib, fetchurl, fetchpatch
 , perl, pkg-config
 , libcap, libtool, libxml2, openssl, libuv
+, enableGSSAPI ? true, libkrb5
 , enablePython ? config.bind.enablePython or false, python3 ? null
```
[the remark this answers is not in the document]
Comment (illustris) [reply pkgs/servers/dns/bind/default.nix:5]: oh nevermind, I just noticed that enablePython is right there

**Comment by fabianhjr** [comment]
Result of nix-review:

```
3 packages blacklisted:
tests.nixos-functions.nixos-test tests.nixos-functions.nixosTest-test tests.trivial

28 packages built:
acme-sh autofs5 bashSnippets bind blueberry check-wmiplus cinnamon.cinnamon-common cinnamon.cinnamon-gsettings-overrides cinnamon.cinnamon-screensaver cinnamon.cinnamon-session cinnamon.cjs cinnamon.nemo cinnamon.warpinator cinnamon.xapps cinnamon.xviewer dig dwm-status host inxi monitoring-plugins nmapsi4 python38Packages.xapp python39Packages.xapp sssd testssl twa xed-editor xplayer
```

**Comment by illustris** [comment]
> Result of nix-review on NixOS x86_64:
> 
> ```
> 3 packages blacklisted:
> tests.nixos-functions.nixos-test tests.nixos-functions.nixosTest-test tests.trivial
> 
> 28 packages built:
> acme-sh autofs5 bashSnippets bind blueberry check-wmiplus cinnamon.cinnamon-common cinnamon.cinnamon-gsettings-overrides cinnamon.cinnamon-screensaver cinnamon.cinnamon-session cinnamon.cjs cinnamon.nemo cinnamon.warpinator cinnamon.xapps cinnamon.xviewer dig dwm-status host inxi monitoring-plugins nmapsi4 python38Packages.xapp python39Packages.xapp sssd testssl twa xed-editor xplayer
> ```

Looks about right. Most of these are because of the change needed in bind.

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 143648 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>28 packages built:</summary>  <ul>    <li>acme-sh</li>    <li>autofs5</li>    <li>bashSnippets</li>    <li>bind</li>    <li>blueberry</li>    <li>check-wmiplus</li>    <li>cinnamon.cinnamon-common</li>    <li>cinnamon.cinnamon-gsettings-overrides</li>    <li>cinnamon.cinnamon-screensaver</li>    <li>cinnamon.cinnamon-session</li>    <li>cinnamon.cjs</li>    <li>cinnamon.nemo</li>    <li>cinnamon.warpinator</li>    <li>cinnamon.xapps</li>    <li>cinnamon.xviewer</li>    <li>dig</li>    <li>dwm-status</li>    <li>host</li>    <li>inxi</li>    <li>monitoring-plugins (nagiosPluginsOfficial)</li>    <li>nmapsi4</li>    <li>python38Packages.xapp</li>    <li>python39Packages.xapp</li>    <li>sssd</li>    <li>testssl</li>    <li>twa</li>    <li>xed-editor</li>    <li>xplayer</li>  </ul></details>

**Pull request merged**